pkg/build/generic/docker.go

```diff
@@ -0,0 +1,100 @@
+package generic
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/testground/testground/pkg/api"
+	"github.com/testground/testground/pkg/build/common"
+	"github.com/testground/testground/pkg/docker"
+	"github.com/testground/testground/pkg/rpc"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+)
+
+const (
+	buildNetworkName = "testground-build"
+)
+
+var (
+	_ api.Builder = &DockerGenericBuilder{}
+)
+
+type DockerGenericBuilder struct {
+	Enabled bool
+}
+
+type DockerGenericBuilderConfig struct {
+	BuildArgs    map[string]*string `toml:"build_args"`
+	PushRegistry bool               `toml:"push_registry"`
+	RegistryType string             `toml:"registry_type"`
```

Comment:
It's OK to get this in, but it becomes more apparent that this shouldn't be in the builders. How would you feel about working on https://github.com/testground/testground/issues/644 straight after we've merged this? We can then remove the push from here and from `docker:go`.

Reply:
That's for sure. Re-implementing the image-push logic in two builders builders is just rediculous! Yeah I can take a stab at https://github.com/testground/testground/issues/644 after this.